### PR TITLE
docs: remove async-function

### PR DESCRIPTION
**Pull request by atian25**
##### Checklist

- [ ] `npm test` passes
- [ ] tests and/or benchmarks are included
- [ ] documentation is changed or added
- [ ] commit message follows commit guidelines

### CODE REVIEW

File: docs/source/zh-cn/migration.md [hunk 1]
```diff
@@ -177,6 +179,7 @@ const { news, user } = await app.toPromise(ctx.service.biz.list(topic, uid));
 - [toAsyncFunction][app.toAsyncFunction] 和 [toPromise][app.toPromise] 实际使用的是 [co] 包装，因此会带回对应的性能损失和堆栈问题，建议开发者还是尽量全链路升级。
 - [toAsyncFunction][app.toAsyncFunction] 在调用 async function 时不会有损失。
 
+@sindresorhus 编写了许多[基于 promise 的 helper 方法](https://github.com/sindresorhus/promise-fun)，灵活的运用它们配合 async function 能让代码更加具有可读性。
```
Comment (atian25) [inline docs/source/zh-cn/migration.md]: async 那篇一些内容移到这边来了

File: docs/source/zh-cn/intro/egg-and-koa.md [hunk 2]
```diff
@@ -128,11 +128,11 @@ exports.handler = ctx => {
 
 #### Egg 1.x
 
-Egg 1.x 发布时，Node.js 的 LTS 版本尚不支持 async function，所以 Egg 1.x 仍然基于 Koa 1.x 开发，但是在此基础上，Egg 全面增加了 async function 的支持，再加上 Egg 对 Koa 2.x 的中间件也完全兼容，应用层代码可以完全基于 [async function 来实现](../tutorials/async-function.md)。
+Egg 1.x 发布时，Node.js 的 LTS 版本尚不支持 async function，所以 Egg 1.x 仍然基于 Koa 1.x 开发，但是在此基础上，Egg 全面增加了 async function 的支持，再加上 Egg 对 Koa 2.x 的中间件也完全兼容，应用层代码可以完全基于 `async function` 来开发。
 
 - 底层基于 Koa 1.x，异步解决方案基于 [co] 封装的 generator function。
 - 官方插件以及 Egg 核心使用 generator function 编写，保持对 Node.js LTS 版本的支持，在必要处通过 co 包装以兼容在 async function 中的使用。
-- 应用开发者可以选择 async function（Node.js 7.6+） 或者 generator function（Node.js 6.0+）进行编写，**我们推荐 generator function 方案以确保应用可以运行在 Node.js LTS 版本上**。
+- 应用开发者可以选择 async function（Node.js 7.6+） 或者 generator function（Node.js 6.0+）进行编写。
```
Comment (popomore) [inline docs/source/zh-cn/intro/egg-and-koa.md]: Node 8 吧

Comment (atian25) [reply docs/source/zh-cn/intro/egg-and-koa.md]: 改了

Comment (popomore) [reply docs/source/zh-cn/intro/egg-and-koa.md]: 还有英文的

Comment (atian25) [reply docs/source/zh-cn/intro/egg-and-koa.md]: done

**Comment by popomore** [comment]
有个链接挂了

**Comment by atian25** [comment]
fixed

**Comment by atian25** [comment]
合了?

**Pull request merged**